### ISSUE TITLE
Problem with xstring

### ISSUE THREAD

**Issue by luchete80** [luchete80/WeldForm#63]
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\xstring(549): error C2988: declaración o definición de plantilla irreconocible
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\xstring(549): error C2143: error de sintaxis: falta ';' delante de '<'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\xstring(549): error C2059: error de sintaxis: '<'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\xstring(549): error C2065: '_Elem': identificador no declarado
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\xstring(549): error C2923: 'std::char_traits': '_Elem' no es un argumento de tipo de plantilla vá
```

**Comment by luchete80** [comment]
https://developercommunity.visualstudio.com/t/visual-studio-2019-compilation-issue/525532

> Thanks for your feedback! In order to investigate this issue, could you please provide a cpp file which could reproduce your issue?
> 
> We are looking forward to your reply!
> 
> 
> 
> 0
> 
> 
> Apr 11, 2019
> avatar
> 
>  
> Solution - 
> Jared Burkeen  
> ···
> I was able to resolve the issue by adding the header <sstream> to the affected file.
> 
> It's interesting that the same file compiled fine with VS 2017. I ran into another similar issue and the fix was to add the missing include file.
> 
> 
> 
> 1
> 
> 
> 
> 
> Show comments 0
> Apr 26, 2019
> avatar
>   
> Cheney Wang [MSFT]  
> Closed - Not a Bug
> ···
> Thank you for your feedback! We have determined that this issue is by design. In order to improve throughput, we have removed the iostreams stuff from string in VS2019, this makes it so that users don't get all of the iostreams stuff whenever they touch anything that uses <string>.

**Comment by luchete80** [comment]
https://stackoverflow.com/questions/58228789/undefined-type-and-undeclared-identifiers-in-xstring

**Comment by luchete80** [comment]
https://social.msdn.microsoft.com/Forums/vstudio/pt-BR/2a97c4dc-f1d0-4beb-b7d5-70bef1958e79/getting-compilation-errors-after-upgraded-a-c-project-from-vs2013-to-vs2017-rc?forum=vclanguage

**Comment by luchete80** [comment]
> Thank you everyone for help! I found the reason of the issue: there was the compiler option
> 
>   Expand Attribute Source Yes (/Fx)
> 
> As soon as I set the option to No, project started compiling successfully. Last time we used the option was in VS2005, and then we forgot about it. Only VS2017 breaks when it is on. This could be a bug in VS2017. Could you check?
> 
> I reproduced the issue in an empty Console C++ App. When stdafx.h includes STL headers, and /Fx was used.

**Comment by luchete80** [comment]
THIS IS THE PROBLEM:

https://stackoverflow.com/questions/531916/error-c2039-memchr-is-not-a-member-of-global-namespace/531928

Ok I fixed it myself. It was a stupid mistake! I have a file called "String.h" in a library project which is being picked up by the <cstring> header. Probably because I have added the path to <String.h> as an additional include directory in my test project (where I am getting this error.) Hope this helps someone.
